### PR TITLE
 Log DBR BuildInfo [databricks]

### LINKED ISSUE #8587: Improve Databricks runtime  shim detection
We currently rely on the prefix of the version strings

https://github.com/NVIDIA/spark-rapids/blob/8b752454ca27557bcfebc5a98d459f039ee5bee0/sql-plugin/src/main/spark332db/scala/com/nvidia/spark/rapids/shims/spark332db/SparkShimServiceProvider.scala#L34-L36

whose values are documented in the [spark-versions API](https://docs.databricks.com/api/workspace/clusters/sparkversions). These versions represent wildcards for the latest patch of the major.minor such as `11.3.x`.

Thus, a user of an older rapids-4-spark artifact may hit a runtime bug or worse, a silent defect, instead of a clear actionable message as implemented in #8521 

Spark UI on DBR displays "Build Properties"   in the Environment Tab:

Name  | Value
-- | --
Runtime Build Hash | 383fa9ccdbf99891a97ff2c546d4330d923a6d82
Universe Build Hash | e3f8b198b7c7c313f95719b7f41d3503780d4a4d

These values correspond to

```scala
org.apache.spark.BuildInfo.gitHash
com.databricks.BuildInfo.gitHash
```
in a Scala notebook

which we can be utilized in the patch version detection.

### PR DESCRIPTION
**Pull request by gerashegalov**
Fixes #8587
 
- Match the dbrVersion from the binaries
- Log the build info exposed in current_version
- Fix the WITH_BLOOP build

```
2024-09-10 16:21:42,019 [Thread-6] WARN  com.nvidia.spark.rapids.DatabricksShimServiceProvider - Databricks Runtime Build Info match: SUCCESS
        DBR_VERSION: 11.3.x-snapshot-gpu-ml-scala2.12
        spark.BuildInfo.gitHash: d65fb2374451fd10bf416297dc22549bcbaf2702
        databricks.BuildInfo.gitHash: b7fd9d058866e0f98f78304bf90c690198e2b208
        databricks.BuildInfo.gitTimestamp: 20240820204043
```
    
Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/DatabricksShimServiceProvider.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                  |\tspark.gitHash: {}
+                  |\tdatabricks.gitHash: {}""".stripMargin,
+        matchStatus,
+        sparkBuildInfo.dbrVersion,
+        sparkBuildInfo.gitHash,
+        databricksBuildInfo.gitHash)
+      matchRes
+    }.recover {
+      case x: Throwable =>
+        log.warn("Databricks detection failed: " + x, x)
```
Comment (jlowe) [inline sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/DatabricksShimServiceProvider.scala:45]: Should this be a debug log?  IIUC we're going to iterate through the shims, calling matchesVersion on each, until we find one that matches.  We don't want a warning appearing on startup for a non-Databricks Spark version, and ShimLoader itself will only log debug when having issues loading any particular shim.

File: sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/DatabricksShimServiceProvider.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+object DatabricksShimServiceProvider {
+  val log = org.slf4j.LoggerFactory.getLogger(getClass().getName().stripSuffix("$"))
+  def matchesVersion(dbrVersion: String): Boolean = {
+    Try {
+      val sparkBuildInfo = org.apache.spark.BuildInfo
+      val databricksBuildInfo = com.databricks.BuildInfo
+      val matchRes = sparkBuildInfo.dbrVersion.startsWith(dbrVersion)
+      val matchStatus = if (matchRes) "SUCCESS" else "FAILURE"
+      log.warn(s"""Databricks Runtime Build Info match: {}
```
Comment (jlowe) [inline sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/DatabricksShimServiceProvider.scala:34]: Same log question here -- do we want to log messages on shims that aren't appropriate but just visited as part of the shim detection?  Seems like this should only be logged on success (which removes the need for the first line of the log), with debug logging for failure cases.

File: sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/spark330db/SparkShimServiceProvider.scala
Comment (jlowe) [inline sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/spark330db/SparkShimServiceProvider.scala:1]: 2024 copyrights

File: sql-plugin/src/main/spark332db/scala/com/nvidia/spark/rapids/shims/spark332db/SparkShimServiceProvider.scala
Comment (jlowe) [inline sql-plugin/src/main/spark332db/scala/com/nvidia/spark/rapids/shims/spark332db/SparkShimServiceProvider.scala:1]: 2024 copyrights

File: sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/spark341db/SparkShimServiceProvider.scala
Comment (jlowe) [inline sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/spark341db/SparkShimServiceProvider.scala:1]: 2024 copyrights

**Comment by gerashegalov** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**